### ISSUE TITLE
Support of PHP heredoc and nowdoc syntax for autocomplete and calltip display

### ISSUE THREAD

**Issue by seattledeveloper** [Komodo/KomodoEdit#784]
I'd like to suggest more functionality in addition to the color scheme support as described in https://github.com/Komodo/KomodoEdit/issues/761.

As the title notes, it would be great if autocomplete and calltips displayed within HTML inside of heredocs and nowdocs.  In fact, this should really be inside all strings (both single and double quoted).  As an aside, my old editor provided this functionality.

My current workaround is to copy the text I'm editing out of the PHP part of the file, edit it, and then copy it back in when done.  Pretty awkward.

Thanks in advance for considering this enhancement!

Ed

**Comment by Naatan** [comment]
Could you possible give a visual/textual example of what your workflow is, so we have a better understanding of what you are asking for? Thanks!

**Comment by seattledeveloper** [comment]
Hi Nathan,

Here's an example of HTML written outside of a PHP block of code with the tip text showing:

![tiptext-outside-php-block](https://cloud.githubusercontent.com/assets/9749337/10892548/e8a9b786-8158-11e5-9460-2225a8d1ab5d.png)

Here's an example of the &lt;div tag not being recognized as HTML, nor tiptext displayed, when tag opened inside a heredoc:

![heredoc-with-no-tiptext](https://cloud.githubusercontent.com/assets/9749337/10892594/380f2c0c-8159-11e5-8d13-71a47bf19ab9.png)

And here's the same example inside a string declaration:

![string-constant-with-no-tiptext](https://cloud.githubusercontent.com/assets/9749337/10892618/4abc189c-8159-11e5-9499-91839da191bf.png)

And here's a screen shot from Homesite, which had been my editor until I recently switched to Komodo:

![homesite_screenshot](https://cloud.githubusercontent.com/assets/9749337/10892722/f91aae58-8159-11e5-836b-fcf728fedba7.png)

In summary, my ask is that all the html in a file (whether or not in php blocks of code) is treated the same, both in coloring (from my earlier request), and in autocomplete and calltip.

Hope this is helpful.  Please let me know if you'd like any further information.

Ed

**Comment by Naatan** [comment]
Gotcha, thanks for explaining :)

**Comment by Naatan** [comment]
I've renamed the ticket you linked to, that will track the support of heredoc, which includes the color scheme and codeintel.
